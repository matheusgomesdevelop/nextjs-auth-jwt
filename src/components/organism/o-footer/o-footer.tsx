// StyleSheet
import styles from "./o-footer.module.scss";

// Atoms
import { A_Icon } from "@/components/atoms/a-icon/a-icon";
import { A_Link } from "@/components/atoms/a-link/a-link";

export interface I_OFooter {
  name: string;
  site: string;
  github: string;
  linkedin: string;
}

export const O_Footer = ({ name, site, github, linkedin }: I_OFooter) => {
  const date = new Date();

  return (
    <footer className={styles.o_footer}>
      <div className={styles.o_footer__name}>
        <p>
          Â© {date.getFullYear()}
          <A_Link href={site}>{name}</A_Link>.
        </p>
      </div>

      <div className={styles.o_footer__icons}>
        <A_Link href={github}>
          <A_Icon icon="pi-github" />
        </A_Link>

        <A_Link href={linkedin}>
          <A_Icon icon="pi-linkedin" />
        </A_Link>
      </div>
    </footer>
  );
};
