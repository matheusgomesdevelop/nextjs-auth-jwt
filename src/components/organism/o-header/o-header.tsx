import styles from "./o-header.module.scss";

// Atoms
import { A_Avatar } from "@/components/atoms";

export const O_Header = () => {
  return (
    <header data-testid="o-header" className={styles.o_header}>
      <A_Avatar image={"/images/a-avatar.jpeg"} label={""} shape="circle" />
    </header>
  );
};
