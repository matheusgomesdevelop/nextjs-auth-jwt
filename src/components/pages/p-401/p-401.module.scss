@import "../../../styles/util/_breakpoints.scss";

.p_401 {
  width: 100%;
  height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;

  & > [id="t-401"] {
    padding: 0 var(--spacing-16);
    margin-top: -64px;

    @include d(tablet) {
      padding: 0;
      margin-top: -96px;
    }
  }
}
