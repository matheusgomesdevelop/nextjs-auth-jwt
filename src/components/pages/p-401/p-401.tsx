export interface I_P401 {
  t_401Props: {
    handleLogin: () => void;
  };
}

export const P_401 = ({ t_401Props }: I_P401) => {
  return (
    <main data-testid="p-401">
      <T_401 />
    </main>
  );
};
