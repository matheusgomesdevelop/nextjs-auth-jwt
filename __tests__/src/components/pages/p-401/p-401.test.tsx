import renderer from "react-test-renderer";

// Interface
import { I_OHeader } from "@/components/organism/o-header/o-header";
import { I_OFooter } from "@/components/organism/o-footer/o-footer";

// Page
import { P_401 } from "@/components/pages/p-401/p-401";

describe("Deve renderizar a pagina 401 corretamenta", () => {
  // Organism: Header
  const o_headerProps: I_OHeader = {
    link: "https://github.com/matheusgrodrigues",
    image: "/images/a-avatar.jpeg",
  };

  // Organism: Footer
  const o_footerProps: I_OFooter = {
    name: "matheusgomesdev",
    site: "https://matheusgomesdev.com.br",
    github: "https://github.com/matheusgrodrigues",
    linkedin: "https://www.linkedin.com/in/matheusgomes/",
  };

  it("Deve preservar a estrutura da pagina p-401", () => {
    const three = renderer.create(
      <P_401
        o_headerProps={o_headerProps}
        t_401Props={{
          handleLogin: () => null,
        }}
        o_footerProps={o_footerProps}
      />
    );

    expect(three).toMatchSnapshot();
  });
});
