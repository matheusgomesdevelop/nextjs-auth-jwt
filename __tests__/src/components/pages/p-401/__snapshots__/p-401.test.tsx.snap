// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar a pagina 401 corretamenta Deve preservar a estrutura da pagina p-401 1`] = `
<main
  className="p_401"
  data-testid="p-401"
>
  <header
    className="o_header"
    data-testid="o-header"
    id="o-header"
  >
    <a
      className="a_link"
      href="https://github.com/matheusgrodrigues"
    >
      <div
        className="p-avatar p-component p-avatar-image p-avatar-circle"
        data-pc-name="avatar"
        data-pc-section="root"
        data-testid="a-avatar"
        style={{}}
      >
        <img
          alt="avatar"
          data-pc-section="image"
          onError={[Function]}
          src="/images/a-avatar.jpeg"
        />
      </div>
    </a>
    <button
      className="a_button p-button p-component"
      data-pc-name="button"
      data-pc-section="root"
      data-testid="a-button"
      disabled={false}
      onClick={[Function]}
    >
      <i
        className="a_icon pi pi-sun"
      />
    </button>
  </header>
  <main
    className="t_401"
    data-testid="t-401"
    id="t-401"
  >
    <p
      className="a_text__fwSb_fs16_primary"
      data-testid="a-text"
    >
      401
    </p>
    <h1
      className="a_title__h1"
      data-testid="a-title"
    >
      Acesso não autorizado
    </h1>
    <p
      className="a_text__fwReg_fs16_gray500"
      data-testid="a-text"
    >
      Desculpe, você não tem permissão para acessar esta página. Por favor, verifique suas credenciais ou entre em contato com o suporte se precisar de assistência.
    </p>
    <button
      className="a_button__gradient p-button p-component"
      data-pc-name="button"
      data-pc-section="root"
      data-testid="a-button"
      disabled={false}
      onClick={[Function]}
    >
      Faça Login
    </button>
  </main>
  <div
    className="o_footer"
    id="o-footer"
  >
    <div
      className="o_footer__name"
    >
      <p>
        © 
        2023
        <a
          className="a_link"
          href="https://matheusgomesdev.com.br"
        >
          matheusgomesdev
          .
        </a>
      </p>
    </div>
    <div
      className="o_footer__icons"
    >
      <a
        className="a_link"
        href="https://github.com/matheusgrodrigues"
      >
        <i
          className="a_icon pi pi-github"
        />
      </a>
      <a
        className="a_link"
        href="https://www.linkedin.com/in/matheusgomes/"
      >
        <i
          className="a_icon pi pi-linkedin"
        />
      </a>
    </div>
  </div>
</main>
`;
